# HG changeset patch
# User Rik <rik@octave.org>
# Date 1522104849 25200
#      Mon Mar 26 15:54:09 2018 -0700
# Branch stable
# Node ID 92252c8c755966479288e36c9a2b15a3c5a88268
# Parent  11ab997e76cea51f2e10cb3d3c184ad51b02da0d
mk_undocumented_list: Check intermediate results for sanity before continuing.

* mk_undocumented_list: Change regexp to grab any characters after '=' and use
as the location.  Do a sanity check after parsing Octave script output to
see if the number of functions matches the number of locations.  Add new
exceptions to the list of functions which require documentation.

diff --git a/doc/interpreter/doccheck/mk_undocumented_list b/doc/interpreter/doccheck/mk_undocumented_list
--- a/doc/interpreter/doccheck/mk_undocumented_list
+++ b/doc/interpreter/doccheck/mk_undocumented_list
@@ -42,17 +42,23 @@ unless (@octave_output) { die "Unable to
 # First, divide output in to list of functions and list of locations
 $idx = 0;
 while (($_ = $octave_output[$idx++]) !~ /^#!-separator-!#$/)
 {
   push(@all_functions, $1) if (/] = (\w+)$/);
 }
 while ($_ = $octave_output[$idx++])
 {
-  push(@where, $1) if (/] = (\S*)$/);
+  push(@where, $1) if (/] = (.+)$/);
+}
+
+# Sanity check that Octave script worked
+if ($#all_functions != $#where)
+{
+  die "Unequal number of functions and locations.  Parsing failed\n"; 
 }
 
 # Second, remove functions based on directory location
 # deprecated directory, doc/interpreter directory, test/ directory
 FUNC: foreach $idx (0 .. $#where)
 {
   next FUNC if ($where[$idx] =~ /deprecated/i);
   next FUNC if ($where[$idx] =~ /interpreter/i);
@@ -103,39 +109,40 @@ exit(1);
 angle
 bessel
 besselh
 besseli
 besselk
 bessely
 bug_report
 chdir
+dbnext
 debug
-dbnext
 end
 exit
 F_DUPFD
 F_GETFD
 F_GETFL
+F_SETFD
+F_SETFL
 fact
 finite
 flipdim
 fmod
-F_SETFD
-F_SETFL
 gammaln
 home
 i
 ifelse
+import
 inf
 inverse
 isbool
 isfinite
+J
 j
-J
 java2mat
 lower
 lstat
 metaclass
 nan
 nargchk
 O_APPEND
 O_ASYNC
@@ -143,19 +150,19 @@ O_CREAT
 O_EXCL
 O_NONBLOCK
 O_RDONLY
 O_RDWR
 O_SYNC
 O_TRUNC
 O_WRONLY
 putenv
-SEEK_CUR
-SEEK_END
 setenv
-tmpnam
-toc
-triu
+slash
 upper
 ylabel
 ylim
+yticklabels
+yticks
 zlabel
 zlim
+zticklabels
+zticks
